### PR TITLE
 opts: Allow string concatenation (fixes #7199) 

### LINKED ISSUE #7199: meson_options.txt don't allow splitting long strings nicely
Currently `meson_options.txt` don't provide a way to split long strings in multiple lines. If I try to use the `+` operator, the syntax is rejected. My use case is the `description` value. Descriptions can be long to be useful, but I don't like writing anything beyond the 80-column barrier and would rather break this long line.

### PR DESCRIPTION
**Pull request by mensinda**

### CODE REVIEW

File: docs/markdown/snippets/options_string_concat.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+## String concatenation in meson_options.txt
+
+It is not possible to use string concatenation (with the `+` opperator) in the
```
Comment (xclaesse) [inline docs/markdown/snippets/options_string_concat.md:3]: Not->now

**Comment by xclaesse** [comment]
Otherwise looks good to me

File: docs/markdown/snippets/options_string_concat.md (hunk 1 above)
Comment (mensinda) [reply docs/markdown/snippets/options_string_concat.md]: Fixed, Thanks.

**Comment by TheQwertiest** [comment]
Why not use python\C\C++ syntax instead? I.e. when two consequent strings are treated as one :
```
option(
  'test' 'option',
  type : 'string',
  value : 'optval',
  description : 'An option with a very long description'
                      'that does something in a specific context'
)
```

**Comment by mensinda** [comment]
Because you can't do that in plain meson either and `meson_options.txt` just uses a subset of normal meson code (which means that it also heavily relies on the existing AST / interpreter).

**Comment by TheQwertiest** [comment]
Then it should be added in meson interpreter as well :P Joking (or am I? :D)

**Comment by mensinda** [comment]
Sure, but not in this PR :)

**Comment by xclaesse** [comment]
LGTM.

**Pull request merged**